### PR TITLE
[ONNX]Support exporting squeeze & unsqueeze with negative dim attribute

### PR DESCRIPTION
**Pull request by BowenBao**

### CODE REVIEW

**Comment by BowenBao** [comment]
cc @houseroad

File: torch/onnx/symbolic.py [hunk 1]
```diff
@@ -555,6 +555,14 @@ def squeeze(g, self, dim=None):
                 dims.append(i)
     else:
         dims = [_get_const(dim, 'i', 'dim')]
+        # Handle negative dims
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: actually, before we call type().sizes(), we need to check whether we have the type information of the tensor: something like `if input.type().kind() == "CompleteTensorType" or input.type().kind() == "DimensionedTensorType":`. Check how we did it in softmax case.

Comment (BowenBao) [reply torch/onnx/symbolic.py]: Thanks! updated with the check

File: test/onnx/test_pytorch_onnx_caffe2.py [hunk 2]
```diff
@@ -990,7 +990,10 @@ def test_convtranspose(self):
 
     def test_unsqueeze(self):
         shape = (3, 4, 5)
-        for dim in range(len(shape) + 1):
+        for dim in range(2 * len(shape) + 2):
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_caffe2.py]: for dim in range(-len(shape), len(shape), 1)

File: test/onnx/test_pytorch_onnx_caffe2.py [hunk 3]
```diff
@@ -1000,6 +1003,21 @@ def forward(self, x):
             x = torch.randn(*shape)
             self.run_model_test(MyModel(), train=False, input=(x), batch_size=BATCH_SIZE, atol=1e-7)
 
+    def test_squeeze(self):
+        shape = (1, 1, 1)
+        for dim in range(2 * len(shape)):
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_caffe2.py]: ditto

File: torch/onnx/symbolic.py [hunk 4]
```diff
@@ -555,6 +555,17 @@ def squeeze(g, self, dim=None):
...
+        for i, dim in enumerate(dims):
+            if dim < 0:
+                if self.type().kind() == "CompleteTensorType" or self.type().kind() == "DimensionedTensorType":
+                    warnings.warn("ONNX export squeeze with negative axis " + str(dim) +
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: Why warning? Because we suppose the rank won't change at runtime? If so, please make it clear in the warning.

File: torch/onnx/symbolic.py [hunk 5]
```diff
@@ -1383,6 +1394,16 @@ def alias(g, self):
 
 @parse_args('v', 'i')
 def unsqueeze(g, self, dim):
+    # Handle negative dim
+    if dim < 0:
+        if self.type().kind() == "CompleteTensorType" or self.type().kind() == "DimensionedTensorType":
+            warnings.warn("ONNX export squeeze with negative axis " + str(dim) +
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: ditto.

**Comment by houseroad** [comment]
Please address the comments and rebase the PR. Thanks!

File: torch/onnx/symbolic.py (hunk 4 above)
Comment (BowenBao) [reply torch/onnx/symbolic.py]: Thanks! done

**Review by houseroad — APPROVED** [review]
Looks good to me.
